Bottom: 45b79191abed4ca3ba19e01fbb7df93bd87c0385
Top:    744323dde8cad00e65af5de501341f35c5e0fbec
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2012-03-30 20:17:22 -0400

Only set the stop bit on the last message



---

diff --git a/drivers/i2c/busses/i2c-pnx.c b/drivers/i2c/busses/i2c-pnx.c
index 855c534..585971a 100644
--- a/drivers/i2c/busses/i2c-pnx.c
+++ b/drivers/i2c/busses/i2c-pnx.c
@@ -219,7 +219,8 @@ static int i2c_pnx_master_xmit(struct i2c_pnx_algo_data *alg_data)
 		/* We still have something to talk about... */
 		val = *alg_data->mif.buf++;
 
-		if (alg_data->mif.len == 1)
+		/* last byte of a message */
+		if ((alg_data->mif.len == 1) && alg_data->last)
 			val |= stop_bit;
 
 		alg_data->mif.len--;
